### PR TITLE
Add image-pushing job for capo

### PR DESCRIPTION
**Pull request by sbueringer**
This PR adds a job to push CAPO images similar to the CAPI ones: https://github.com/kubernetes/test-infra/pull/14195

Hope this PR is not to wrong. I'mt not sure what the correct process is or if there's anything else to do :)

/cc @Katharine @detiber

### CODE REVIEW

**Comment by Katharine** [comment]
This looks correct to me. You'll want kubernetes-sigs/cluster-api-provider-openstack#460 in first. That PR looks plausible but I can't really review because I don't know how your build process works - that part is repo specific. GCB does provide instructions for [testing on your machine](https://cloud.google.com/cloud-build/docs/build-debug-locally) which might be helpful to you.

You'll also need a bucket created and permissions to be set up on your staging GCR, which should happen once kubernetes/k8s.io#356 merges and is run by someone (assuming @thockin didn't do something already).

/hold

**Comment by detiber** [comment]
@sbueringer might want to incorporate similar changes as this PR as well: https://github.com/kubernetes/test-infra/pull/14228/files

**Comment by sbueringer** [comment]
@Katharine Thx. I verified the job locally, worked perfect. So I make sure to get the PR in CAPO merged and wait until the other one from @thockin.

**Comment by sbueringer** [comment]
> @sbueringer might want to incorporate similar changes as this PR as well: https://github.com/kubernetes/test-infra/pull/14228/files

Thx I saw the update in CAPI but missed this one here :)

File: config/jobs/image-pushing/k8s-staging-capi-openstack.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        - ^master$
+      spec:
+        containers:
+          - image: gcr.io/k8s-testimages/image-builder:v20190906-d5d7ce3
+            command:
+              - /run.sh
+            args:
+              # this is the project GCB will run in, which is the same as the GCR images are pushed to.
+              - --project=k8s-staging-capi-openstack
+              - --scratch-bucket=gs://k8s-staging-capi-openstack-scratch
```
Comment (Katharine) [inline config/jobs/image-pushing/k8s-staging-capi-openstack.yaml]: This name has changed to `k8s-staging-capi-openstack-gcb`.

```suggestion
              - --scratch-bucket=gs://k8s-staging-capi-openstack-gcb
```

**Comment by sbueringer** [comment]
@Katharine thx :), fixed

**Comment by Katharine** [comment]
Feel free to unhold when relevant changes have landed in your repo.

**Comment by sbueringer** [comment]
> Feel free to unhold when relevant changes have landed in your repo.

Changes have been merged to the CAPO repo. Basically the same as in the CAPI repo, so I guess it should work :)

/hold cancel

**Pull request merged**